store/tikv/snapshot_test.go

```diff
@@ -119,23 +119,30 @@ func (s *testSnapshotSuite) TestBatchGet(c *C) {
 	}
 }
 
+type key string
+
+const (
+	x key = "x"
+	y key = "y"
+)
+
 func (s *testSnapshotSuite) TestSnapshotCache(c *C) {
 	txn := s.beginTxn(c)
-	c.Assert(txn.Set(kv.Key("x"), []byte("x")), IsNil)
-	c.Assert(txn.Delete(kv.Key("y")), IsNil) // store data is affected by other tests.
+	c.Assert(txn.Set(kv.Key(x), []byte(x)), IsNil)
```

Comment:
The change makes me confused.
What kind of error message does the staticcheck tool report?

Reply:
should not use built-in type string as key for value; define your own type to avoid collisions (SA1029)

Reply:
```
cd store/tikv;
staticcheck
```

It doesn't report SA1029 but some other ones:

```
➜  tikv git:(Rustin-Liu/rustin-patch-store) staticcheck
2pc.go:798:3: the surrounding loop is unconditionally terminated (SA4004)
client.go:135:4: grpc.WithBackoffMaxDelay is deprecated: use WithConnectParams instead. Will be supported throughout 1.x.  (SA1019)
region_cache.go:85:2: should use copy() instead of a loop (S1001)
region_cache.go:432:23: func (*RegionCache).loadAndInsertRegion is unused (U1000)
store_test.go:111:24: func (*mockPDClient).enable is unused (U1000)
txn.go:74:2: field assertions is unused (U1000)
txn.go:75:2: field confirmed is unused (U1000)
txn.go:106:6: type assertionPair is unused (U1000)
```

@Rustin-Liu

Reply:
I will fix that after this pr, this pr only for store.

Reply:
Have you ever notice here in `snapshot_test.go`
https://github.com/pingcap/tidb/blob/de4b456f17358ceb39734652882fb943956b9351/store/tikv/snapshot_test.go#L220

There is also a `c.Assert(txn.Set(kv.Key("x"), []byte("x")), IsNil)` and is there any error report for it?

Reply:
Now store still have these static check issues:
region_cache.go:430:23: func (*RegionCache).loadAndInsertRegion is unused (U1000)
store_test.go:111:24: func (*mockPDClient).enable is unused (U1000)
txn.go:74:2: field assertions is unused (U1000)
txn.go:75:2: field confirmed is unused (U1000)
txn.go:106:6: type assertionPair is unused (U1000)